flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/sink/shuffle/SortKeySerializer.java

```diff
@@ -310,10 +324,16 @@ public void writeSnapshot(DataOutputView out) throws IOException {
     @Override
     public void readSnapshot(int readVersion, DataInputView in, ClassLoader userCodeClassLoader)
         throws IOException {
-      if (readVersion == 1) {
-        readV1(in);
-      } else {
-        throw new IllegalArgumentException("Unknown read version: " + readVersion);
+      switch (readVersion) {
+        case 1:
+          throw new UnsupportedOperationException(
+              String.format(
+                  "No longer supported version [%s]. Please upgrade first . ", readVersion));
+        case 2:
+          readV2(in);
+          break;
```

Comment:
Reading the snapshot isn't any different for V1/V2. The `readV1()` or `readV2()` methods are identical. We can just keep the current one. However, we need to save the serializer version as an `int` field which we can read in `restoreSerializer()`, just like the schema and the sort order. We can then serialize/deserialize depending on the version used.

For example, in the serializer we would do:
```
if (version == 1) {
  // Don't read/write null indicator
} else {
  // Read/write null indicator
}